src/ucp/api/ucp.h

```diff
@@ -333,6 +335,19 @@ typedef enum {
 
 /**
  * @ingroup UCP_DATATYPE
+ * @brief Generate a reusable identifier for Scatter-gather IOV data type.
```

Comment:
i think this kind of optimization could be done for any datatype transparently without special API
same way we have registration cache optimization, for example. no need to ask for it explicitly.